### PR TITLE
Improve aws lambda deployment (logging, idempotency, etc)

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: This PR fixes or improves a couple of things
1. Docker push command was failing for me without any errors in Python, which took quite some time to debug. With this change I make it possible to print out docker command outputs. This is logged on the debug level, so you'll need to run it with `--log-level=debug`, for example `feast --log-level=debug apply`. Also, looks like log-level flag was not working correctly, so I also fixed that.
2. We shouldn't try docker image download/upload if the image is already in ECR. So I made a change that first checks this image in ECR and only runs the docker logic if necessary. This also means that from the 2nd `feast apply`, users won't need to have docker daemon running (which was required on each apply before).

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/infra/aws.py [hunk 1]
```diff
@@ -244,52 +255,72 @@ def _upload_docker_image(self, project: str) -> str:
...
-    def _get_version_for_aws():
-        """Returns Feast version with certain characters replaced.
+def _get_lambda_name(project: str):
+    lambda_prefix = AWS_LAMBDA_FEATURE_SERVER_REPOSITORY
+    lambda_suffix = f"{project}-{_get_version_for_aws()}"
+    # AWS Lambda name can't have the length greater than 64 bytes.
+    # This usually occurs during integration tests or when feast is
+    # installed in editable mode (pip install -e), where feast version is long
+    if len(lambda_prefix) + len(lambda_suffix) >= 63:
+        lambda_suffix = base64.b64encode(lambda_suffix.encode()).decode()[:40]
```
Comment (felixwang9817) [inline sdk/python/feast/infra/aws.py:314]: is 40 enough here? looks like the prefix is 36 characters?

**Review by felixwang9817 — APPROVED** [review]
looks great overall, just one quick comment

File: sdk/python/feast/infra/aws.py (hunk 1 above)
Comment (adchia) [reply sdk/python/feast/infra/aws.py:314]: isn't the prefix feast-python-server which is 19 characters? so 19+40 = 59 < 63.

**Review by adchia — APPROVED** [review]
will leave Felix to give the lgtm, but the code lgtm

File: sdk/python/feast/infra/aws.py (hunk 1 above)
Comment (felixwang9817) [reply sdk/python/feast/infra/aws.py:314]: oops it is

**Pull request merged**